python/lsst/ts/ofc/scripts/generate_alpha_values.py

```diff
@@ -0,0 +1,271 @@
+        )
+        yaml_file.write(header)
+        yaml.dump(alpha_values_list, yaml_file)
+
+
+def get_alpha_values(
+    config: dict,
+    instrument: str,
+    band: str,
+    znmin: int = 4,
+    znmax: int = 28,
+) -> np.ndarray:
+    """Calculate alpha values for the normalized point-source sensitivity.
+
+    Parameters
+    ----------
+    config : dict
+        Configuration dictionary.
+    instrument : str
+        Instrument name.
+    band : str
+        Filter band.
+    znmin : int
+        Minimum Zernike index.
+    znmax : int
+        Maximum Zernike index.
+
+    Returns
+    -------
+    np.ndarray
+        Alpha values for the normalized point-source sensitivity.
+    """
+
+    # Set wavlength and number of zernikes used
+    num_zk = znmax - znmin + 1
+    wavelength = config["wavelengths"][band]
+
+    # Set the increments of zernikes used to do the fit
+    increments = np.array([2, 4, 6, 8, 10]) * wavelength * 1e-2
+
+    # Create OFCData, OPDMetrology and fiducial telescope
+    ofc_data = OFCData("lsst")
+    opd_metrology = OpdMetrology()
+
+    if band == "ref":
+        fiducial = batoid.Optic.fromYaml(f"{instrument.upper()}_g_500.yaml")
+    else:
+        fiducial = batoid.Optic.fromYaml(f"{instrument.upper()}_{band}.yaml")
+
+    # Get the field angles of the gaussian quadrature points
+    field_angles = [
+        ofc_data.gq_points[sensor] for sensor in range(len(ofc_data.gq_weights))
+    ]
+    field_x, field_y = zip(*field_angles)
+
+    # Load the gaussian quadrature weights
+    imqw = [ofc_data.gq_weights[sensor] for sensor in range(len(ofc_data.gq_weights))]
+    n_imqw = imqw / np.sum(imqw)
+
+    # Create empty arrays to store the pssn values
+    pssn_avg = np.zeros((num_zk, len(increments)))
+    pssn_gq_points = np.zeros((num_zk, len(field_angles), len(increments)))
+    alpha_values = np.zeros(num_zk)
+
+    for id_zk in tqdm(range(num_zk)):
+        zk = np.zeros(25)
+
+        # For each increment, compute the PSSN values
+        for id_increment, increment in enumerate(increments):
+            # Set value of zernike perturbation
+            zk[id_zk] = increment
+
+            for id_angle in range(len(field_angles)):
+                # Convert field angles to radians
+                field_x = np.deg2rad(field_angles[id_angle][0])
+                field_y = np.deg2rad(field_angles[id_angle][1])
+
+                # Calculate the wavefront without perturbation
+                # to retrieve the opd mask
+                opd_w = batoid.wavefront(
+                    fiducial, field_x, field_y, wavelength=wavelength, nx=255
+                )
+                opd_size = opd_w.array.shape[0]
+                opd_grid_1d = np.linspace(-1, 1, opd_size)
+                opd_x, opd_y = np.meshgrid(opd_grid_1d, opd_grid_1d)
+                non_masked = ~opd_w.array.mask
+
+                # Set zernikes for the new opd
+                new_opd = copy(opd_w.array)
```

Comment:
what is the size/shape of `opd_w.array`? I am not sure you need to do `copy()` or `deepcopy()` here? If this `array` is a `numpy.array`, there should be a `copy()` function already.